From 2a34db55059236e16d62a85c9dbceed7b5ce28b2 Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Mon, 1 Jul 2019 17:44:54 +0200
Subject: [PATCH 12/15] Do not list / load dependencies if `vagrant` spec is
 not loaded

in `vagrant_internal_specs` as this fails, due to `find` returning `nil`.
---
 lib/vagrant/bundler.rb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/vagrant/bundler.rb b/lib/vagrant/bundler.rb
index 7ba48435f..c0fabdcea 100644
--- a/lib/vagrant/bundler.rb
+++ b/lib/vagrant/bundler.rb
@@ -421,8 +421,9 @@ module Vagrant
     def vagrant_internal_specs
       # activate any dependencies up front so we can always
       # pin them when resolving
-      Gem::Specification.find { |s| s.name == "vagrant" && s.activated? }.
-        runtime_dependencies.each { |d| gem d.name, *d.requirement.as_list }
+      if (vs = Gem::Specification.find { |s| s.name == "vagrant" && s.activated? })
+        vs.runtime_dependencies.each { |d| gem d.name, *d.requirement.as_list }
+      end
       # discover all the gems we have available
       list = {}
       directories = [Gem::Specification.default_specifications_dir]
-- 
2.22.1

